src/main/java/org/candlepin/paging/FieldComparatorFactory.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2009 - 2023 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.paging;
+
+import java.util.Comparator;
+
+
+
+/**
+ * A functional interface for defining a mapping function that converts a field name to a comparator
+ * for the purposes of sorting collections or streams for paged output.
+ *
+ * @param <T>
+ *  the class type for which this factory's mapping logic applies
+ */
+@FunctionalInterface
+public interface FieldComparatorFactory<T> {
+
+    /**
+     * Fetches a comparator which orders elements of this comparator factory's type by using the
+     * value returned by accessor best matching the given field name. The comparator this method
+     * returns should always order elements in ascending order. If the type supported by this
```

Comment:
Why ascending when the default is descending?

Reply:
To keep the comparator logic decoupled from our API convention. The PagingUtil will reverse the comparator as necessary based on whatever the request needs.